engine/src/main/java/com/google/android/fhir/search/MoreSearch.kt

```diff
@@ -113,6 +116,25 @@ fun StringFilter.query(type: ResourceType): SearchQuery {
   )
 }
 
+/**
+ * Extension function that returns a SearchQuery based on the value and prefix of the NumberFilter
+ */
+fun NumberFilter.query(type: ResourceType): SearchQuery {
+  val value = value!!
+
+  val precision = value.getRange()
+
+  val conditionParamPair = getConditionParamPair(prefix, value, precision)
```

Comment:
```suggestion
  val conditionAndParams = getConditionParamPair(prefix, value!!)
```

and call `value.getRange()` inside `getConditionParamPair`